lib/handlers/formatting.js

```diff
@@ -25,104 +25,86 @@
 import _ from 'underscore';
 
 import * as exec from '../exec';
+import * as all from '../formatters/_all';
 import { logger } from '../logger';
 
-export class Formatter {
+export class FormattingHandler {
     constructor(ceProps) {
-        this.tools = {};
+        this.formatters = {};
         this.ceProps = ceProps;
         const formatters = _.compact(ceProps('formatters', '').split(':'));
-        _.each(formatters, this.fetchToolInfo.bind(this));
+        _.each(formatters, this.getFormatterInfo.bind(this));
     }
 
-    async fetchToolInfo(formatter) {
+    async getFormatterInfo(formatter) {
         const exe = this.ceProps(`formatter.${formatter}.exe`);
+        const type = this.ceProps(`formatter.${formatter}.type`);
         if (!exe) {
-            logger.warn(`Formatter ${formatter} does not have a valid executable. Skipping...`);
-            return;
+            return logger.warn(`Formatter ${formatter} does not have a valid executable. Skipping...`);
+        }
+        if (!type) {
+            return logger.warn(`Formatter ${formatter} does not have a formatter class. Skipping...`);
         }
         const versionArg = this.ceProps(`formatter.${formatter}.version`, '--version');
         const versionRe = this.ceProps(`formatter.${formatter}.versionRe`, '.*');
         try {
-            const result = await this.run(exe, [versionArg], {});
+            const result = await exec.execute(exe, [versionArg], {});
             const match = result.stdout.match(versionRe);
-            this.tools[formatter] = {
+            const HandleClass = Object.values(all).find((key) => key.key === type);
```

Comment:
You should not have to implement this by hand - please use `makeKeyedTypeGetter` and export a function in the missing index.js I mentioned above.